ballerina/modules/apex/client.bal

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/jballerina.java;
+import ballerinax/'client.config;
+import ballerinax/salesforce.utils;
+
+# Ballerina Salesforce Apex Client provides the capability to access Salesforce Apex REST API.
+# This client allows you to perform operations for custom Apex REST endpoints, execute HTTP methods on these endpoints,
+# and handle responses appropriately.
+public isolated client class Client {
+    private final http:Client salesforceClient;
+    private map<string> sfLocators = {};
+
+    # Initializes the Salesforce Apex Client. During initialization, you can pass either `http:BearerTokenConfig`
+    # if you have a bearer token or `http:OAuth2RefreshTokenGrantConfig` if you have OAuth tokens.
+    # Create a Salesforce account and obtain tokens following 
+    # [this guide](https://help.salesforce.com/articleView?id=remoteaccess_authenticate_overview.htm).
+    #
+    # + config - Salesforce connector configuration
+    # + return - `error` on failure of initialization or else `()`
+    public isolated function init(ConnectionConfig config) returns error? {
```

Comment:
Can't we use included record params here ?
```ballerina
    public isolated function init(*ConnectionConfig config) returns error? {
        // implement the init
    }
```

Reply:
We will add this as a future enhancement.